source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -104,8 +104,29 @@ void CacheFilter::onHeaders(LookupResult&& result) {
       state_ = GetHeadersState::GetHeadersResultUnusable;
     }
     return;
-  case CacheEntryStatus::SatisfiableRange: // TODO(#10132): break response content to the ranges
-                                           // requested.
+  case CacheEntryStatus::NotSatisfiableRange:
+    result.headers_->setStatus(static_cast<uint64_t>(Http::Code::RangeNotSatisfiable));
+    result.headers_->setContentLength(0);
+    result.headers_->addCopy(Http::Headers::get().ContentRange,
+                             absl::StrCat("bytes */", result.content_length_));
+    result.content_length_ = 0;
+  case CacheEntryStatus::SatisfiableRange:
+    // Multi-part responses are not supported; they will be treated as usual 200 response on '::Ok'
+    // case below. A possible way to achieve that would be to move all ranges to remaining_body, and
+    // add logic inside '::onBody' to interleave the body bytes with sub-headers and separator
+    // string for each part. Would need to keep track if the current range is over or not to know
+    // when to insert the separator, and calculate the length based on length of ranges + extra
+    // headers and separators.
+    if (result.response_ranges_.size() == 1) {
+      remaining_body_ = std::move(result.response_ranges_);
+      result.headers_->setStatus(static_cast<uint64_t>(Http::Code::PartialContent));
+      result.headers_->setContentLength(remaining_body_[0].length());
+      result.headers_->addCopy(Http::Headers::get().ContentRange,
+                               absl::StrCat("bytes ", remaining_body_[0].begin(), "-",
+                                            remaining_body_[0].end() - 1, "/",
+                                            result.content_length_));
+      result.content_length_ = remaining_body_[0].length();
+    }
```

Comment:
Another break/return/ABSL_FALLTHROUGH_INTENDED needed. (This one I suspect might be a fallthrough?)

Reply:
Yep, should be a fallthrough.
We just add the range that we want to respond, and the following case will take care of finishing up.